xdo id hangs indefinitely

New to bspwm so I'm sure this is something I've missed; however, I attempted running the example panel as outlined in the popular [bspwm for dummies](https://github.com/windelicato/dotfiles/wiki/bspwm-for-dummies), and after discovering that I also needed xdo & installed it, the panel hung. It turns out to be this command on [this line](https://github.com/baskerville/bspwm/blob/master/examples/panel/panel#L3):

```
xdo id -a "$PANEL_WM_NAME"
```

The env var above is set to "bspwm_panel". Running that command in xterm within bspwm (or even with X off) just hangs.

Any thoughts, or any insight into how to debug this?

Ah; I missed a compile time warning that explains it perfectly:

```
xdo.c: In function 'main':
xdo.c:60:69: warning: comparison is always true due to limited range of data type [-Wtype-limits]
     while ((opt = getopt(argc, argv, "rcCdDsn:N:a:p:k:t:x:y:h:w:")) != -1) {
```

I'm compiling on a Raspberry Pi 3 -- armv7

--- a/xdo.c
+++ b/xdo.c
@@ -56,7 +56,7 @@ int main(int argc, char *argv[])
 
     init();
     argc--, argv++;
-    char opt;
+    int opt;
     while ((opt = getopt(argc, argv, "rcCdDsn:N:a:p:k:t:x:y:h:w:")) != -1) {
         switch (opt) {
             case 'r':

